ws-server/src/server.rs

```diff
@@ -213,6 +249,24 @@ impl Builder {
 		self
 	}
 
+	/// Set a list of allowet `Origin` headers, connections comming in with a different
+	/// origin will be denied. Values should include protocol.
+	///
+	/// ```rust
+	/// # let mut builder = jsonrpsee_ws_server::WsServerBuilder::default();
+	/// builder.set_allowed_origins(vec!["https://example.com"]);
+	/// ```
+	///
+	/// By default allows any `Origin`.
+	pub fn set_allowed_origins<Origin, List>(mut self, list: List) -> Self
```

Comment:
I'm ok with the current function myself (I'd just add a note in the comment that setting an empty list will deny all origins). Perhaps if we added an `allow_all_origins` function to toggle to `AllowAny` it would become a little clearer by contrast? This would also allow one to programmatically undo the effect of calling `set_allowed_origins`, which doesn't feel super necessary, but for the sake of completeness it feels nice to be able to set the full range of possible values.

Reply:
(To elaborate a little; if we had an `add_allowed_origins` instead that pushed to the list, it would feel slightly odd to me that you could add an origin to go from allowing everything (nothing added yet) to only allowing what's been added. Whereas, "set_allowed_origins" feels clear to me and doesn't have that slightly odd transition in my mind)

Reply:
My two cents: having a single `set_` method is enough. We do not expect the list to change once the server is started, and even if it is technically possible to dynamically add/remove allowed origins, I think it's fine to be opinionated here.

Wrt to the semantics of setting the allowed origins to the empty set, I think we should pick one of these:
1. empty list => everyone can connect, i.e. switch to `Cors::AllowAny`
2. empty list => return error

Reply:
I'd be more in favour of 2: I wouldn't be super keen on an empty list (perhaps by mistake) loosening restrictions and allowing all origins, so getting an error back feels more helpful.